cdf_fabric_replicator/data_modeling.py

```diff
@@ -339,5 +354,37 @@ def write_instances_to_lakehouse_tables(
                 self.logger.error(f"Error writing instances to lakehouse tables: {e}")
                 raise e
             self.logger.info(
-                f"Successfully wrote {len(instances[table])} to '{abfss_path}' table."
+                f"Successfully wrote {len(instances[table])} rows to '{abfss_path}' table."
+            )
+
+    def delete_instances_from_lakehouse_tables(
+        self, instances: Dict[str, Any], abfss_prefix: str
+    ) -> None:
+        token = self.azure_credential.get_token("https://storage.azure.com/.default")
+        for table in instances:
+            abfss_path = f"{abfss_prefix}/Tables/{table}"
+            self.logger.info(
+                f"Deleting {len(instances[table])} rows from '{abfss_path}' table..."
+            )
+            dt = DeltaTable(
+                abfss_path,
+                storage_options={
+                    "bearer_token": token.token,
+                    "use_fabric_endpoint": str(
+                        self.config.extractor.use_fabric_endpoint
+                    ).lower(),
+                },
+            )
+            try:
+                dt.delete(
```

Comment:
Interestingly enough, there is nothing stopping the external-id to contain a single quote here. Actually, it can contain any kind of valid SQL, and spaces. 

Does the deltalake libraries expose any quote function or similar?